lib/parse.js

```diff
@@ -2467,12 +2467,52 @@ function parse($TEXT, options) {
     };
 
     function is_assignable(expr) {
-        if (!options.strict) return true;
-        if (expr instanceof AST_This) return false;
-        if (expr instanceof AST_Super) return false;
-        return (expr instanceof AST_PropAccess || expr instanceof AST_Symbol);
+        return expr instanceof AST_PropAccess || expr instanceof AST_SymbolRef;
     };
 
+    function to_destructuring(node) {
```

Comment:
https://github.com/mishoo/UglifyJS2/pull/1627#issuecomment-288637954

I've moved the original `var walk = function(node) {...}` out to here.